### PR TITLE
Issue#5 - add form and update addItem function

### LINKED ISSUE #5: 5. As a user, I want to add a new item to my shopping list so I can start recording purchases
## Summary
Users need to be able to add items to their shopping lists. To do this, the `AddItem` view should present them with a form that allows them to enter two things:
1. The name of the item, and
2. An estimate of when they think they’ll need to buy the item again.

[The README file for the shopping list API](https://github.com/the-collab-lab/smart-shopping-list-next/blob/main/src/api/README.md) will be helpful as you work on this feature.

## Acceptance criteria
UI-related tasks:
- [x] The `ManageList` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Notes
- When this feature is implemented correctly, new items will *automatically* show up in the list view because `streamListItem` gets new data from Firestore every time there’s a change.
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- The add item form has the potential to be a lot of scope. Stick to the Acceptance Criteria.

### PR DESCRIPTION
**Pull request by amandaguan-ag**
## Description
Added a form to ManageList page and updated addItem function in firebase component

## Related Issue

Closes #5 

## Acceptance Criteria
UI-related tasks:
- [x] The `ManageList` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Type of Changes

Update addItem feature

### Before

image

### After
image

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull orign ag-gl-add-item-functionality` and check that branch out with `git checkout ag-gl-add-item-functionality` 
* Start development server with `npm start`
* Fill out the form with a new item and its purchase urgency. Make sure it is in the list page and `Your item was added!` message shows up

### CODE REVIEW

**Review by chungleee — APPROVED** [review]
Everything works on my end! 

I will note on my first try that adding an item didn't work. I typed the item name, saw that `soon` was the default so I didn't click on dropdown but it didn't work. I think setting `soon` as default value would correct that and save the user an extra click :D maybe i'm being nitpicky.

**Review by RachaelSkye — APPROVED** [review]
Looks great and works as expected. I really appreciate the error handling. Nice work!

File: src/views/ManageList.jsx [hunk 1]
```diff
@@ -1,7 +1,71 @@
...
+				nextPurchasedDate = 14;
+				break;
+			case 'notSoon':
+				nextPurchasedDate = 30;
+				break;
+			default:
+				console.log('Unrecognized selection');
+				return;
+		}
+		try {
```
Comment (djtaylor8) [inline src/views/ManageList.jsx:31]: Nice usage of try/catch!

File: src/views/ManageList.jsx [hunk 2]
```diff
@@ -1,7 +1,71 @@
...
-		<p>
-			Hello from the <code>/manage-list</code> page!
-		</p>
+		<div>
+			<p>
+				Hello from the <code>/manage-list</code> page!
+			</p>
+			{submitted === 'added' && <span>Your item was added!</span>}
+			{submitted === 'failed' && <span>Your item wasn't added!</span>}
+			<form onSubmit={handleSubmit}>
```
Comment (djtaylor8) [inline src/views/ManageList.jsx:50]: I mentioned this in the other pull request, but thanks for keeping this form simple. Again, be thinking about potential styles, etc you all would want to add later.

File: src/views/ManageList.jsx [hunk 3]
```diff
@@ -1,7 +1,71 @@
...
+			default:
+				console.log('Unrecognized selection');
+				return;
+		}
+		try {
+			await addItem(listPath, {
+				itemName: name,
+				daysUntilNextPurchase: nextPurchasedDate,
+			});
+			setSubmitted('added');
```
Comment (djtaylor8) [inline src/views/ManageList.jsx:36]: suggestion - can we reset the form input fields after an item has been successfully added? Not necessary right now, but might be a nice user experience.

**Review by djtaylor8 — APPROVED** [review]
Great job @amandaguan-ag and @g-lee2! Just a few minor comments. Again, apologies for the Firestore mixup this week.

**Review by lyleschemmerling — APPROVED** [review]
lgtm

**Pull request merged**